docs/internals/development.rst

```diff
@@ -0,0 +1,55 @@
+Developer Information
+=====================
+
+The following information is intended for developers who wish to modify the
+code.
+
+The pages on :doc:`testing` and :doc:`documentation` are also relevant.
+
+Building BridgeStan
+-------------------
+
+Developers should follow the instructions in :doc:`../getting-started` to ensure
+they have the minimal C++ build tools required for working with BridgeStan.
+
+C++
+---
+
+* We use the C++1y standard for compilation (``-std=c++1y`` in both clang and gcc). This is partway between C++11 and C++14, and is what Stan requires.
+
+* We try to write standards-compliant code that does not depend on features of specific platforms (except where needed for compatibility).
+  Specifically, we do not use OS-dependent or compiler-dependent C++.
+  Our C++ code does not depend on the ``R.h`` or ``Python.h`` headers, for example.
+  On the other hand, adding new signatures to work with a specific language's style of foreign function interface is permitted
+  (an example can be found in the :ref:`R compatibility functions <R-compat>`, which requires a particular pointer-based style).
+
+* We try to follow the `Google C++ Style Guide <https://google.github.io/styleguide/cppguide.html>`_, but (a) we allow C++ exceptions, and (b) we allow reference arguments.
```

Comment:
There seems to be a dangling underscore after the URL.   Or is that just indicating some kind of space?  We definitely don't want an underscore but I also don't think we want a space between a formatted URL and punctuation.

Reply:
That's just how links look in [RST](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#hyperlinks)

Reply:
I looked it up and wow, they really do include the underscore to mean it's an inline link.  So I retract all those comments about not wanting spaces---I had thought it was just GitHub highlighting a space :-).